.github/workflows/snyk-scan.yml

```diff
@@ -0,0 +1,171 @@
+name: Snyk Security Vulnerability Scan
+
+on:
+  pull_request:
+  push:
+    branches:
+      - 'master'
+      - 'rel-*'
+
+jobs:
+  snyk_scan_test:
+    if: ${{ github.event_name == 'pull_request' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0 # To fetch all commits history on branch (Refer: https://github.com/tj-actions/changed-files#usage)
+       
+      - name: Check changed Deps files
+        uses: tj-actions/changed-files@v10
+        id: changed-files
+        with:
+          files: | # This will match all the files with below patterns
+            build.gradle$
+            requirements.txt$
+            package.json$
+
+      - uses: snyk/actions/setup@master
+
+      - uses: actions/setup-java@v1
+        with:
+          java-version: "8"
+          distribution: 'adopt'
+      - name: Snyk scan for Java dependancies 
+        if: contains(steps.changed-files.outputs.all_changed_and_modified_files, 'build.gradle')
+        id: scan1
+        continue-on-error: true
+        run: |
+          unset CI # By default GH actions will set it to true. Therfore it will affect isCi flag in build.gradle (line #7)
+          snyk test --all-sub-projects -d --fail-on=all --package-manager=gradle
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+
+      - uses: actions/setup-python@v1
+        with:
+          python-version: "3.5"
+      - name: Snyk scan for Python 3.5 dependancies
+        if: contains(steps.changed-files.outputs.all_changed_and_modified_files, 'requirements.txt')
+        id: scan2
+        continue-on-error: true
+        env:
+          SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+        run: |
+          echo "##[set-output name=result;]$(echo success)" # set default
+          sudo apt-get install -y libkrb5-dev
+          pip install -r h2o-docs-theme/requirements.txt
+          pip install -r h2o-py/docs-requirements.txt  
```

Comment:
What is the purpose of the scan? Do we need to scan stuff that we do not ship to customers?

Reply:
mainly the ultimate goal is to scan the stuff we ship to customers. we can remove this step if required since it is related to documentation

Reply:
Removed the docs-related deps files from the Snyk scan. Please review the new changes when you have time and let me know if there's anything we can omit further